tests/actions/ReportTest.js

```diff
@@ -0,0 +1,102 @@
+        const TEST_USER_ACCOUNT_ID = 1;
+        const TEST_USER_LOGIN = 'test@test.com';
+        const REPORT_ID = 1;
+        const ACTION_ID = 1;
+        const REPORT_ACTION = {
+            actionName: 'ADDCOMMENT',
+            actorAccountID: TEST_USER_ACCOUNT_ID,
+            actorEmail: TEST_USER_LOGIN,
+            automatic: false,
+            avatar: 'https://d2k5nsl2zxldvw.cloudfront.net/images/avatars/avatar_3.png',
+            isAttachment: false,
+            isFirstItem: false,
+            loading: true,
+            message: [{type: 'COMMENT', html: 'Testing a comment', text: 'Testing a comment'}],
+            person: [{type: 'TEXT', style: 'strong', text: 'Test User'}],
+            sequenceNumber: ACTION_ID,
+            shouldShow: true,
+            timestamp: moment().unix(),
+        };
+
+        // When using the Pusher mock the act of calling Pusher.isSubscribed will create a
+        // channel already in a subscribed state. These methods are normally used to prevent
+        // duplicated subscriptions, but we don't need them for this test.
+        Pusher.isSubscribed = jest.fn().mockReturnValue(false);
+        Pusher.isAlreadySubscribing = jest.fn().mockReturnValue(false);
+
+        // Connect to Pusher
+        PusherConnectionManager.init();
+        Pusher.init({
+            appKey: CONFIG.PUSHER.APP_KEY,
+            cluster: CONFIG.PUSHER.CLUSTER,
+            authEndpoint: `${CONFIG.EXPENSIFY.URL_API_ROOT}api?command=Push_Authenticate`,
+        });
+
+        let reportActions;
+        Onyx.connect({
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${REPORT_ID}`,
+            callback: val => reportActions = val,
+        });
+
+        // Set up Onyx with some test user data
+        return signInWithTestUser(TEST_USER_ACCOUNT_ID, TEST_USER_LOGIN)
+            .then(() => {
+                subscribeToReportCommentEvents();
+                return waitForPromisesToResolve();
+            })
+            .then(() => fetchPersonalDetailsForTestUser(TEST_USER_ACCOUNT_ID, TEST_USER_LOGIN, {
+                [TEST_USER_LOGIN]: {
+                    accountID: TEST_USER_ACCOUNT_ID,
+                    email: TEST_USER_LOGIN,
+                    firstName: 'Test',
+                    lastName: 'User',
+                },
+            }))
+            .then(() => {
+                // This is a fire and forget response, but one it completes we should be able to verify that we
+                // have an "optimistic" report action in Onyx.
+                addAction(REPORT_ID, 'Testing a comment');
+                return waitForPromisesToResolve();
+            })
+            .then(() => {
+                const resultAction = reportActions[ACTION_ID];
+                expect(resultAction).toEqual(REPORT_ACTION);
+            })
+            .then(() => {
+                // Now that we are subscribed we need to simulate a reportComment action Pusher event.
```

Comment:
> `// Now that we are subscribed`

Subscribed to what? Is this referring to the original Pusher subscription or to Onyx? This reads a little weird because I would expect the thing before this to be code that does subscribing, but the thing before this is adding the report action.

Reply:
I can make this more explicit.

Reply:
It probably made more sense before when the thing immediately before this was `subscribeToReportComments()`